packages/sanity/src/desk/components/deskTool/DeskToolBoundary.tsx

```diff
@@ -20,15 +17,6 @@ export function DeskToolBoundary({tool: {options}}: DeskToolBoundaryProps) {
   const [firstSource] = sources
   const {source, defaultDocumentNode, structure} = options || {}
 
-  const {state: routerState} = useRouter()
-  const intent = useMemo(() => {
-    const intentName = typeof routerState.intent === 'string' ? routerState.intent : undefined
-    const params = isRecord(routerState.params) ? routerState.params : EMPTY_RECORD
-    const payload = routerState.payload
-
-    return intentName ? {intent: intentName, params, payload} : undefined
-  }, [routerState])
-
```

Comment:
We don't need this anymore? Wondering about it since looking at the comment on line 46 🤔 
My assumption is that we don't need it because we are resolving this at a different time*?

Reply:
Yes, this data was only used by `<IntentResolver>` so I moved it into that component. This reduces the re-render burden for `DeskToolBoundary` since it no longer has to re-render on every single URL update 😌

Reply:
Really cool 🙇